connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -274,9 +275,9 @@ object KustoWriter {
                                 ingestionProperties: IngestionProperties,
                                 partitionsResults: CollectionAccumulator[PartitionResult]): util.ArrayList[Future[Unit]]
   = {
+    val partitionId = TaskContext.getPartitionId
```

Comment:
Small comment again , Since we are using toString multiple times , may be this we can create the string here and use it
    val partitionIdString = TaskContext.getPartitionId.toString